code/__HELPERS/game.dm

```diff
@@ -217,31 +217,28 @@
 
 	return
 
-/proc/get_hearers_in_view(R, atom/source)
-	// Returns a list of hearers in view(R) from source (ignoring luminosity). Used in saycode.
+///Returns a list of hearers in view(R) from source (ignoring luminosity). Used in saycode.
+/proc/get_hearers_in_view(range, atom/source)
 	var/turf/T = get_turf(source)
-	. = list()
-	if(!T)
-		return
-	var/list/processing_list = list()
-	if (R == 0) // if the range is zero, we know exactly where to look for, we can skip view
-		processing_list += T.contents // We can shave off one iteration by assuming turfs cannot hear
+
+	if (range == 0) // if the range is zero, we know exactly where to look for, we can skip view
+		. = T.contents.Copy() // We can shave off one iteration by assuming turfs cannot hear
 	else  // A variation of get_hear inlined here to take advantage of the compiler's fastpath for obj/mob in view
+		. = list()
 		var/lum = T.luminosity
 		T.luminosity = 6 // This is the maximum luminosity
-		for(var/mob/M in view(R, T))
-			processing_list += M
-		for(var/obj/O in view(R, T))
-			processing_list += O
+		for(var/mob/M in view(range, T))
+			. += M
+		for(var/obj/O in view(range, T))
+			. += O
 		T.luminosity = lum
 
 	var/i = 0
-	while(i < length(processing_list)) // recursive_hear_check inlined here
-		var/atom/A = processing_list[++i]
+	while(i < length(.)) // recursive_hear_check inlined here
+		var/atom/A = .[++i]
 		if(A.flags_1 & HEAR_1)
-			. += A
-			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, processing_list, .)
-		processing_list += A.contents
+			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, .)
```

Comment:
```suggestion
			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, .)
		else
			. -= A
			i--
```
`get_hearers_in_view` is supposed to return a list of movables with HEAR_1 enabled and no more, comsig aside.